### DISCUSSION TITLE
Working with Panel component and esc button

### DISCUSSION THREAD

**Discussion by adnanjalloh** [microsoft/fluentui#24519] [Q&A] [answered]
Hi all, 

How could I disable the ESC button to close the panel or prevent the user from closing it by pressing the ESC button is there a way to do that?

**Comment by micahgodbolt** [comment, marked as answer]
Since you can control the state of the panel (isOpen) you can look at the key coming in from the onDismiss, and be selective about when you dismiss.

```
onDismiss={(ev) =>  {
  if (ev.key === 'escape') { .... }
}
```

**Reply by adnanjalloh** [reply]
Many thanks
